src/wp-includes/blocks.php

```diff
@@ -114,12 +114,13 @@ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
 			substr_replace( $script_path, '.asset.php', - strlen( '.js' ) )
 		)
 	);
-	if ( ! file_exists( $script_asset_path ) ) {
+	if ( false === $script_asset_path ) {
 		_doing_it_wrong(
 			__FUNCTION__,
 			sprintf(
 				/* translators: 1: Field name, 2: Block name. */
-				__( 'The asset file for the "%1$s" defined in "%2$s" block definition is missing.' ),
+				__( 'The asset file (%1$s) for the "%2$s" defined in "%3$s" block definition is missing.' ),
+				dirname( $metadata['file'] ) . '/' . substr_replace( $script_path, '.asset.php', - strlen( '.js' ) ),
```

Comment:
Could we avoid having to do this twice? A third variable in addition to `$script_handle`, and `$script_asset_path` could store the result of this line before passing to `wp_normalize_path( `realpath() )` above and then to the `_doing_it_wrong()`.

Reply:
I agree. https://github.com/WordPress/wordpress-develop/pull/2313/commits/749efcfd8880b2de069eac862c255127069ad94d stores the result in a variable for reuse.